src/js/lib/safe.js

```diff
@@ -159,26 +166,47 @@ Pebble.sendAppMessage = function(message, success, failure) {
 /* Wrap setTimeout and setInterval */
 var originalSetTimeout = setTimeout;
 window.setTimeout = function(callback, delay) {
+  if (safe.warnSetTimeoutNotFunction !== false && typeof callback !== 'function') {
+    safe.warn('setTimeout was called with a `' + (typeof callback) + '` type. ' +
+              'Did you mean to pass a function?');
+    safe.warnSetTimeoutNotFunction = false;
+  }
   return originalSetTimeout(safe.protect(callback), delay);
 };
+
 var originalSetInterval = setInterval;
 window.setInterval = function(callback, delay) {
+  if (safe.warnSetIntervalNotFunction !== false && typeof callback !== 'function') {
+    safe.warn('setInterval was called with a `' + (typeof callback) + '` type. ' +
+              'Did you mean to pass a function?');
+    safe.warnSetIntervalNotFunction = false;
+  }
   return originalSetInterval(safe.protect(callback), delay);
 };
 
-/* Wrap the success and failure callback of the ajax library */
-ajax.onHandler = function(eventName, callback) {
-  return safe.protect(callback);
-};
-
 /* Wrap the geolocation API Callbacks */
 var watchPosition = navigator.geolocation.watchPosition;
 navigator.geolocation.watchPosition = function(success, error, options) {
   return watchPosition.call(this, safe.protect(success), safe.protect(error), options);
 };
+
 var getCurrentPosition = navigator.geolocation.getCurrentPosition;
 navigator.geolocation.getCurrentPosition = function(success, error, options) {
   return getCurrentPosition.call(this, safe.protect(success), safe.protect(error), options);
 };
 
+var ajax;
+
+/* Try to load the ajax library if available and silently fail if it is not found. */
+try {
+  ajax = require('ajax');
+} catch (err) {}
```

Comment:
pebble.js is a monolithic framework at the moment, but many of its parts can be used in other projects. I want to start modularizing every component. At some point, I would like to be able to recommend using safe.js + loader.js + wscript in someone's C + JS project.